src/control/lib/daos/system_prop.go

```diff
@@ -487,6 +491,50 @@ func pph2sp(key SystemPropertyKey, pph *PoolPropHandler, def string) SystemPrope
 	return property
 }
 
+// SystemPropertySelfHealHasFlag returns true if the given self-heal property
+// value contains the specified flag.
+func SystemPropertySelfHealHasFlag(value string, flag string) bool {
+	if value == "none" {
+		return false
+	}
+
+	flags := strings.Split(value, ";")
+	for _, f := range flags {
+		if f == flag {
+			return true
+		}
+	}
+
+	return false
+}
+
+// subsets returns a slice of all subsets of strings, including the empty set
+// represented by empty. In each nonempty subset, the elements are separated by
+// sep. For example,
+//
+//	subsets([]string{"a","b","c"}, ";", "none")
+//
+// returns
+//
+//	[]string{"a;b;c", "a;b", "a;c", "a", "b;c", "b", "c", "none"}.
+func subsets(strings []string, sep string, empty string) []string {
```

Comment:
Nice! This might be useful to put in our common/collection_utils.go and unit test on its own.